proxy/src/main/java/com/velocitypowered/proxy/connection/client/LoginSessionHandler.java

```diff
@@ -198,6 +201,12 @@ private EncryptionRequest generateEncryptionRequest() {
   }
 
   private void initializePlayer(GameProfile profile, boolean onlineMode) {
+    // Restore normal timeout for connection.
+    mcConnection.getChannel().pipeline()
```

Comment:
Can you do something like mcConnection.getChannel().pipeline().get(Connections.READ_TIMEOUT)**.setTimeout(server.getConfiguration().getReadTimeout()) so maybe the re-instantiation of ReadTimeoutHandler isn't needed?

Reply:
I don't think such nanooptimization is useful, the gc would catch the non-used object right away.